app/controllers/halls_controller.rb

```diff
@@ -0,0 +1,52 @@
+class HallsController < ApplicationController
+
+    before_action :set_hall, only: [:show, :update, :destroy]
+
+    def index
+        @halls = Hall.all.map do |hall|
```

Comment:
I think at this moment it's enough good, but you can read about serializers in the meantime :wink: 

You can take a look at results like that in Google:
https://dev.to/amnyrrd/json-serializers-in-a-rails-api-5262
https://www.botreetechnologies.com/blog/fast-json-api-serializer-by-netflix-in-ruby-on-rails/
https://github.com/rails-api/active_model_serializers
https://medium.com/@rachid1982fsb/using-active-model-serializer-2fe1ffb42909